pyiceberg/table/update/snapshot.py

```diff
@@ -243,13 +233,8 @@ def _commit(self) -> UpdatesAndRequirements:
         next_sequence_number = self._transaction.table_metadata.next_sequence_number()
 
         summary = self._summary(self.snapshot_properties)
-
-        manifest_list_file_path = _generate_manifest_list_path(
-            location=self._transaction.table_metadata.location,
-            snapshot_id=self._snapshot_id,
-            attempt=0,
-            commit_uuid=self.commit_uuid,
-        )
+        file_name = f"{self.commit_uuid}-m{self._snapshot_id}-a0.avro"
+        manifest_list_file_path = self._transaction._table.metadata_file_location(file_name)
```

Comment:
is this the same as `_generate_manifest_list_path`? 

`_generate_manifest_list_path` was using
```
    return f"{location}/metadata/snap-{snapshot_id}-{attempt}-{commit_uuid}.avro"
```

aside from `{location}/metadata` which we are changing, the rest should stay the same.

Reply:
nit: should we keep `_generate_manifest_list_path` and make it generate the file path after `/metadata`, and then just use `.metadata_file_location()` to include the metadata prefix